src/App.jsx

```diff
@@ -1,65 +1,132 @@
 import BlogTitles from './BlogTitles.jsx';
-import {useState, useEffect} from 'react';
-import AddBlogTitle from './AddBlogTitle.jsx'
-import Search from './Search.jsx'
-import {getItem, setItem} from "./useSemiPersistentState.jsx";
+import { useState, useEffect, useReducer } from 'react';
+import AddBlogTitle from './AddBlogTitle.jsx';
+import Search from './Search.jsx';
+import { getItem, setItem } from './useSemiPersistentState.jsx';
 
-const titleToBlog = (title) => ({id: crypto.randomUUID(), title})
+const titleToBlog = title => ({ id: crypto.randomUUID(), title });
 
-const LOCALSTORAGE_KEY = 'savedBlogTitles'
+const initialTitles = [
+  'Lesson 1.1 Project setup',
+  'Lesson 1.2 React DOM and components',
+  'Lesson 1.3 Form input, props, state and callbacks',
+  'Lesson 1.4 Lifting state, controlled components, props handling',
+  'Lesson 1.5 Side effects, custom hooks, fragments',
+  'Lesson 1.6 Reusable Components, Imperative React',
+  'Lesson 1.7 Asynchronous Data, Conditional Rendering',
+];
+
+const LOCALSTORAGE_KEY = 'savedBlogTitles';
+
+const filterBlogTitles = (blogTitles, searchText) =>
+  blogTitles.filter(blogTitle =>
+    blogTitle.title.toLowerCase().includes(searchText.toLowerCase())
+  );
+
+const stateManagementFunction = (previousState, action) => {
+  switch (action.type) {
+    case 'FINISH_LOADING_BLOG_TITLES':
+      return {
+        ...previousState,
```

Comment:
We will have the easiest time if the value of the state returned from our stateManagementFunction has the same properties after each action is run. This returned object spreads / copies the previousState and overwrites any properties in the next lines with the new property values